server/p2p_receive_transaction.go

```diff
@@ -28,124 +25,81 @@ Incoming Data Object Example:
-		TxID: transaction.TxID(),
+	if len(requestPayload.Hex) == 0 {
+		panic("empty hex after parsing!")
 	}
 
-	// Check signature if: 1) sender validation enabled or 2) a signature was given (optional)
-	if c.SenderValidationEnabled || len(p2pTransaction.MetaData.Signature) > 0 {
-
-		// Check required fields for signature validation
-		if len(p2pTransaction.MetaData.Signature) == 0 {
-			ErrorResponse(w, ErrorInvalidSignature, "missing parameter: signature", http.StatusBadRequest)
-			return
-		} else if len(p2pTransaction.MetaData.PubKey) == 0 {
-			ErrorResponse(w, ErrorInvalidPubKey, "missing parameter: pubkey", http.StatusBadRequest)
-			return
-		}
-
-		// Get the address from pubKey
-		var rawAddress *bscript.Address
-		if rawAddress, err = bitcoin.GetAddressFromPubKeyString(p2pTransaction.MetaData.PubKey, true); err != nil {
-			ErrorResponse(w, ErrorInvalidPubKey, "invalid pubkey: "+err.Error(), http.StatusBadRequest)
-			return
-		}
-
-		// Validate the signature of the tx id
-		if err = bitcoin.VerifyMessage(rawAddress.AddressString, p2pTransaction.MetaData.Signature, response.TxID); err != nil {
-			ErrorResponse(w, ErrorInvalidSignature, "invalid signature: "+err.Error(), http.StatusBadRequest)
-			return
-		}
+	if beefData == nil {
+		panic("empty beef after parsing!")
 	}
 
-	// Create the metadata struct
-	md := CreateMetadata(req, alias, domain, "")
-
-	// Get from the data layer
-	var foundPaymail *paymail.AddressInformation
-	foundPaymail, err = c.actions.GetPaymailByAlias(req.Context(), alias, domain, md)
-	if err != nil {
-		ErrorResponse(w, ErrorFindingPaymail, err.Error(), http.StatusExpectationFailed)
-		return
-	} else if foundPaymail == nil {
-		ErrorResponse(w, ErrorPaymailNotFound, "paymail not found", http.StatusNotFound)
+	var err error
+	if err = c.actions.ExecuteSimplifiedPaymentVerification(req.Context(), beefData); err != nil {
+		ErrorResponse(w, ErrorSimplifiedPaymentVerification, err.Error(), http.StatusExpectationFailed)
 		return
 	}
 
-	// Record the transaction (verify, save, broadcast...)
+	var response *paymail.P2PTransactionPayload
 	if response, err = c.actions.RecordTransaction(
-		req.Context(), p2pTransaction, md,
+		req.Context(), requestPayload.P2PTransaction, md,
 	); err != nil {
 		ErrorResponse(w, ErrorRecordingTx, err.Error(), http.StatusExpectationFailed)
 		return
 	}
 
-	// Set the response
 	writeJsonResponse(w, http.StatusOK, response)
 }
 
-// p2pReceiveBeefTx will receive a P2P transaction in BEEF format
-func (c *Configuration) p2pReceiveBeefTx(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
-	parms := req.URL.Query()
-	beef := parms.Get("beef")
-
-	// TODO: Use those values in future processing
-	_, _ = paymail.DecodeBEEF(beef)
+type p2pPayloadFormat uint
 
-	ErrorResponse(w, ErrorNotImplmented, "Receive BEEF transactions not implemented", http.StatusNotImplemented)
-}
+const (
+	basicP2pPayload p2pPayloadFormat = iota
+	beefP2pPayload
+)
```

Comment:
When I look at the file, I personally like seeing consts on the top, or at least before they are first used, it makes it easier to read the code top-down instead of jumping to the bottom to see the definition. WDYT?

Reply:
Moved up ;-)